urlb5d5be4365_abmargb_oursim_ext_tgz-pJ8-br_edu_ufcg_lsd_oursim_experiments_RemoteNoFScenarioGeneratorJ8/src/br/edu/ufcg/lsd/oursim/experiments/ScenarioEvaluator.java

```diff
@@ -0,0 +1,71 @@
+package br.edu.ufcg.lsd.oursim.experiments;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.commons.io.IOUtils;
+
+public class ScenarioEvaluator {
+
+	static String SCENARIO = "resources/experiments/remote-nof";
+	
+	public static void main(String[] args) throws FileNotFoundException, IOException {
+		
+		File root = new File(SCENARIO);
+		analyzeFolder(root, new StringBuilder());
+		
+	}
+
+	private static void analyzeFolder(File root, StringBuilder stringBuilder) throws FileNotFoundException, IOException {
+		boolean onlyFiles = true;
+		
+		for (File file : root.listFiles()) {
+			if (file.isDirectory()) {
+				StringBuilder strBuilderCopy = new StringBuilder(stringBuilder.toString());
+				strBuilderCopy.append(file.getName()).append(" ");
+				analyzeFolder(file, strBuilderCopy);
+				onlyFiles = false;
+			} else {
+				analyzeFile(file);
+			}
+		}
+		
+		if (onlyFiles) {
+			System.out.println(stringBuilder.toString());
+		}
+	}
+
+	private static void analyzeFile(File file) throws FileNotFoundException, IOException {
+		List<String> lines = IOUtils.readLines(new FileInputStream(file));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor: *FileInputStream*. Currently, there are execution paths that do not contain closure statements, for example, when IOUtils.readLines throws an exception. Ensure the object returned by *FileInputStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).